DOCS-6670-address comments (#199)
mulesoft/docs-api-manager: modules/ROOT/pages/configure-policy-template-source.adoc

= Configuring Template Download Location for Policies
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: policy, CloudFront

> A policy template, which is a deployable JAR file, contains the business logic for a policy. Each type of policy has a different template based on the rules and logic that needs to implemented.

Reviewer: ```suggestion
A policy template, which is a deployable JAR file, contains the business logic for a policy. Each type of policy has a different template based on the rules and logic to be implemented.
```

Author: Done! Thanks!

> Before a policy is applied to an API, the policy template is downloaded. When an application with applied policies is deployed to a runtime, it requests API Manager for the corresponding policies to be applied to the API.

Reviewer: ```suggestion
Before a policy is applied to an API, the policy template is downloaded. When an application with applied policies is deployed to a runtime, the application requests API Manager for the corresponding policies to be applied to the API.
```

Author: Done! Thanks!

> //The runtime to which the application is deployed must have API gateway capabilities enabled using the organization or environment credentials where the API instance is defined. Add this content to the How policies work section.

Reviewer: Is this going in to the deliverable or not?
Also, does this mean the same thing:
```suggestion
//The runtime to which the application is deployed must have API gateway permissions that are the same as those of the organization or environment where the API instance is defined. Add this content to the How policies work section.
```
If you move this, be sure to take out the instruction sentence.

Author: Done! Thanks!

> If the policy templates that are to be applied are not already available in the runtime, API gateway requests API Manager for the URLs from where to download the templates.

Reviewer: ```suggestion
If the required policy templates are not already available in the runtime, API gateway requests API Manager for the location from which to download them. By default, this location is the legacy system, but you can configure the runtime to download templates from a specified source:
```

Author: Done! Thanks!

> By default, the legacy system is used as the download location for the templates. You can configure the runtime to download the templates from a specified source using the following system property:

Reviewer: OMIT per previous suggestion
```suggestion
```

Author: Done! Thanks!

> `anypoint.platform.policy_template_source=<source_value>`

Reviewer: ```suggestion
`anypoint.platform.policy_template_source=_source_value_`
```

Reviewer: I am replacing < > with AsciiDoc formatting for italic, which we should use for user-entry placeholders:

Use monospace italics for variables in user input or system output (except in example code blocks, where they do not work, or for DataWeave variables):

	`This is how you tag regular mono.`  

	`_This is how you tag italic mono_.` >

Author: Done! Thanks!

+

> where:

Reviewer: Do not use "where:" as a transition between a syntax example and value explanations. Use plain text or plain text introducing a list:

"In this syntax, '`external` denotes...and `internal` denotes...

HOWEVER...I don't see either "external" or "internal" in the syntax, so to what does this text apply?

Author: Done! Thanks!

[%header,cols="2*a"]
|===
|Source Value | Description

> |"external" | (Default) Denotes that policy templates are to be retrieved from the legacy system. Use this default setting to maintain compatibility with previously configured firewalls.

Reviewer: Shouldn't "external" and "internal" be formatted as syntax, in mono" `external` and `internal`?

Reviewer: Also, the intro text states the default and the example is supposed to be a way to SPECIFY another (internal?) value, so this probably needs to be reworked to clarify the dichotomy of external/default/legacy system and internal/specified/new system.

Author: Done! Thanks!

|"internal" | Denotes that policy templates are to be retrieved from the new system. This option provides additional reliability due to internal rerouting of the requests when a server becomes unavailable.
|===

> The following code snippet shows an example of how to set the template source using the command line:

Reviewer: instead of a system property?
Consider reworking to cover 1) default source using system property, 2) specified new source using system property, and 3) specified new source using CLI?

Author: I think this example is pretty simple. Don't want to complicate any more.

= Configuring Template Download Location for Policies
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: policy, CloudFront

> [PA: Nahuel what is the relationship between policy and templates and how are they used. We need to set some context as to why these templates are needed. What are these templates and how are they used, why are they needed?]

Reviewer 2: Policies in Mule 4 are composed by three files: The business logic, that resides in a Policy Template (a deployable JAR file that contains the policy implementation), the configuration part (a YAML configuration file, where the policy parameters and metadata are defined) and the template's pom.xml file , which among other things, defines the policy dependencies.

Author: Done! Thanks!

When an application with a tracked API is deployed to a runtime with API gateway capabilities enabled, the runtime requests API Manager for the corresponding policies to be applied to the API.

> *Note:* API gateway capabilities are enabled using the organization or environment credentials where the API instance is defined.

Reviewer 2: *G*ateway

Author: No, the way it is referred to as in docs and marketing was us to document it is is API gateway.

If the policy templates that are to be applied are not already available in the runtime, it requests API Manager for the URLs from where to download the templates.

By default, the legacy system is used as the download location for the templates. You can configure the runtime to download the templates from a specified source using the following system property:

`anypoint.platform.policy_template_source=<source_value>`
+
where:

[%header,cols="2*a"]
|===
|Source Value | Description
|"external" | Denotes that policy templates are to be retrieved from the legacy system. Use this default setting to maintain compatibility with previously configured firewalls.
|"internal" | Denotes that policy templates are to be retrieved from the new system. This option provides additional reliability due to internal rerouting of the requests when a server becomes unavailable.
|===

[PA: Nahuel what are we trying to say here? For 4.2.2, the URLs are provided from cloudfront? Why are we calling out only 4.2.2? What about the other versions?]

> When you start Mule Runtime Engine version 4.2.2, API Manager provides the URLs for downloading the policy templates from the Mule CloudFront service, as shown in the following command line example:

Reviewer 2: In the original reviewed document, this line was removed. Can you remove it and replace it by "Command line example:" title or similar?

Author: I made the change based on the editing standards. We don't introduce titles for examples in that way. I'll let my editor guide me. I think we should be fine as long as the content is technically accurate.

Reviewer 2: It does not matter if it is a title, a Note, or something else. Please remove for good the paragraph "If you start Mule version 4.2.2, as shown in the example, API Manager provides the URLs for downloading the policy templates from the Mule CloudFront service." as it is confusing and misleading.

Author: Done! Thanks!

endif::[]
:keywords: policy, CloudFront

> A policy template is a deployable JAR file that contains the business logic for how a policy must be implemented. Before a policy is applied to an API, the policy template is downloaded and accessed to determine the policy dependencies.

Reviewer 2: The statement here is confusing. "A policy template is a deployable JAR file that contains the business logic for how a policy must be implemented." -> "A policy template is a deployable JAR file that contains its business logic."

Reviewer 2: The statement is incorrect. "Before a policy is applied to an API, the policy template is downloaded and accessed to determine the policy dependencies."
You may remove it completely, or explain what's the difference among JAR, YAML and pom files.

Author: This topic is about the policy template and this sentence is explaining what happens to it once you specify the source. I think we should keep it here. We are not talking about JAR and YAML files as they are irrelevant to this topic. They should be covered in the topic where we discuss policies.

When an application with a tracked API is deployed to a runtime, it requests API Manager for the corresponding policies to be applied to the API.

> The runtime to which the application is deployed must have API gateway capabilities enabled using the organization or environment credentials where the API instance is defined.

Reviewer 2: **G** ateway

> The runtime to which the application is deployed must have API gateway capabilities enabled using the organization or environment credentials where the API instance is defined.

Reviewer 2: Tracked API implies runtime correctly configured. I would put this as a note, or remove it altogether.

Author: Changed it to this as it made more sense:
When an application with applied policies is deployed to a runtime, it requests API Manager for the corresponding policies to be applied to the API.
Let me know.
